hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -278,6 +279,32 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to get cluster %q: %w", req.NamespacedName, err)
 	}
 
+	res, err := r.reconcile(ctx, req, log, hcluster)
+	condition := metav1.Condition{
```

Comment:
Yeah it is slightly unintuitive, but that is because the declaration needs to always happen and it contains the transition timestamp, which should not be the time before we started reconciling